### ISSUE TITLE
Note about the categorical_crossentropy?

### ISSUE THREAD

**Issue by diwadd** [fchollet/keras#6580]
Hi,

I'm not sure but the documentation about categorical_crossentropy seems to be not compatible with the tensorflow one.

I mean just had a look at the documentation about loss functions i. e.
https://keras.io/losses/#available-loss-functions

and at the end there is a note about the categorical_crossentropy.
There is such a sentence:

"Note: when using the categorical_crossentropy loss, your targets should be in categorical format (e.g. if you have 10 classes, the target for each sample should be a 10-dimensional vector that is all-zeros expect for a 1 at the index corresponding to the class of the sample)."

So I understand that the only possible target is e.g. [0, 0, 1, 0, 0]?

But in the tensorflow documentation about softmax_cross_entropy_with_logits  we have (https://www.tensorflow.org/api_docs/python/tf/nn/softmax_cross_entropy_with_logits)

"While the classes are mutually exclusive, their probabilities need not be. All that is required is that each row of labels is a valid probability distribution."

which I understand means that such a target is possible [0.0, 0.75, 0.25, 0.0, 0.0]?

categorical_crossentropy is defined in https://github.com/tensorflow/tensorflow/blob/r1.1/tensorflow/contrib/keras/python/keras/backend.py and indeed returns softmax_cross_entropy_with_logits.

Am I missing something (if yes then I'm really sorry)?

Best wishes
Dawid

**Comment by parag2489** [comment]
From what I understand, if you pass `from_logits = False` (default behavior), then the only possible targets are of the form one-hot encoding i.e.  `[0, 0, 1, 0, 0]`. However, if you pass `from_logits = True`, then Keras just calls TensorFlow `softmax_cross_entropy()` function. There I think a target such as `[0.0, 0.75, 0.25, 0.0, 0.0]` is fine. However, I may be wrong since I did not read its C code to find out how `softmax_cross_entropy()` is implemented in TensorFlow.

**Comment by diwadd** [comment]
@parag2489 I guess your right, I have not noticed the `from_logits` flag but still I would say that the **Note** about categorical_crossentropy in 
https://keras.io/losses/#available-loss-functions

is misleading because it does not mention `from_logits` (that is probably why I got confused).

**Comment by gokceneraslan** [comment]
`from_logits` has nothing to do with targets and one-hot encoding. targets can be continuous in any case. It defines how the output (predictions) are interpreted:

![image](https://cloud.githubusercontent.com/assets/1140359/25938149/91b0148e-362e-11e7-8df0-33d26b4aae4f.png)

**Comment by parag2489** [comment]
Yes, I understand that. I had only one doubt. When `from_logits = False`, we should get minimum value when `pred == target`. When I recalculated it now, I did get the minimum value when `pred == target`. Maybe, I did some mistake when I calculated it earlier.

**Comment by diwadd** [comment]
@gokceneraslan you have used the following `target = [0.2, 0.2, 0.6]` and that is my point. In https://keras.io/losses/#available-loss-functions is is said that `target` should be 

> in categorical format (e.g. if you have 10 classes, the target for each sample should be a 10-dimensional vector that is all-zeros expect for a 1 at the index corresponding to the class of the sample)

I think that this is missleading since from what you say (and what is written in the tensorflow docs) `target` can be any valid probability distribution.

**Comment by gokceneraslan** [comment]
It's just to tell people that instead of integer labels like [2, 3, 1] use [[0,0,1,0], [0,0,0,1], [0,1,0,0]]. If you look up the definition of cross-entropy, it's clear that both output and target are probability distributions.
